<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>PleioVar vignette</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">PleioVar vignette</h1>



<div id="before-we-start" class="section level3">
<h3>Before we start</h3>
<p>This vignette is a tutorial to use PleioVar on your own data.<br />
Before starting, a few points:</p>
<ul>
<li>PleioVar relies on LHC-MR results as input parameters. You will have
to process your pairs of traits through LHC-MR before starting
PleioVar.<br />
</li>
<li>LHC-MR is limited to HapMap3 variants, and so is PleioVar.</li>
<li>LHC-MR and PleioVar are computationally intensive methods, and
exponentially so the more traits you add. Parallel computing is highly
recommended.<br />
</li>
</ul>
</div>
<div id="installation" class="section level3">
<h3>Installation</h3>
<p>First, let’s install devtools and PleioVar (if necessary).</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="st">&quot;devtools&quot;</span> <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[, <span class="st">&quot;Package&quot;</span>]) {</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>) }</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>  <span class="cf">if</span>(<span class="sc">!</span><span class="st">&quot;PleioVar&quot;</span> <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[, <span class="st">&quot;Package&quot;</span>]) {</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a>  devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;martintnr/PleioVar&quot;</span>) }</span></code></pre></div>
<p>Then, let’s install LHC-MR from <a href="https://github.com/LizaDarrous/lhcMR" class="uri">https://github.com/LizaDarrous/lhcMR</a> and its
dependencies.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;MRCIEU/TwoSampleMR&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;GenomicSEM/GenomicSEM&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;LizaDarrous/lhcMR&quot;</span>)</span></code></pre></div>
</div>
<div id="setting-up-the-data" class="section level3">
<h3>Setting up the data</h3>
<p>Let’s create and use as working directory PleioVar_NewData folder
that will be used to store necessary and temporary files.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">&quot;PleioVar_New&quot;</span>, <span class="fu">getwd</span>(), <span class="at">fixed =</span> <span class="cn">TRUE</span>)){</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">&quot;PleioVar_New/&quot;</span>)){<span class="fu">system</span>(<span class="st">&quot;mkdir PleioVar_New&quot;</span>)}</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;PleioVar_New&quot;</span>)}</span></code></pre></div>
<p>Some data (like LD-scores) are required for PleioVar and its
dependencies. Let’s download them from zenodo, and also create a folder
for the future LHC-MR results.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a><span class="fu">system</span>(<span class="st">&quot;wget -O Necessary_data.zip  https://zenodo.org/records/10829652/files/Necessary_data.zip?download=1&quot;</span>)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="fu">system</span>(<span class="st">&quot;unzip Necessary_data.zip -d Necessary_data &amp;&amp; rm Necessary_data.zip&quot;</span>)</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">file.exists</span>(<span class="st">&quot;LHCMR_Results&quot;</span>)){<span class="fu">system</span>(<span class="st">&quot;mkdir LHCMR_Results&quot;</span>)}</span></code></pre></div>
<p>Next, you have to indicate the path of your GWAS data and the traits
you want to process.<br />
For this example, I will use LDL, HDL, and CAD from UKBiobank round
2.<br />
You should use LHC-MR (<a href="https://github.com/LizaDarrous/lhcMR" class="uri">https://github.com/LizaDarrous/lhcMR</a>) guidelines to
format your data.<br />
Please format and rename your GWAS data to <em>Trait</em>.csv,
<em>Trait</em>.csv.gz, or <em>Trait</em>.csv.bgz.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a><span class="co">#system(&quot;mkdir GWAS_data&quot;)</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#system(&quot;wget -O HDL.csv.bgz  https://broad-ukb-sumstats-us-east-1.s3.amazonaws.com/round2/additive-tsvs/30760_irnt.gwas.imputed_v3.both_sexes.varorder.tsv.bgz&quot;)</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#system(&quot;wget -O LDL.csv.bgz  https://broad-ukb-sumstats-us-east-1.s3.amazonaws.com/round2/additive-tsvs/30780_irnt.gwas.imputed_v3.both_sexes.varorder.tsv.bgz&quot;)</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="co">#system(&quot;wget -O CAD.csv.bgz  https://broad-ukb-sumstats-us-east-1.s3.amazonaws.com/round2/additive-tsvs/I25.gwas.imputed_v3.both_sexes.tsv.bgz&quot;)</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="co">#system(&quot;mv *.bgz  GWAS_data/&quot;)</span></span></code></pre></div>
</div>
<div id="running-lhc-mr-on-all-pairs-of-traits" class="section level3">
<h3>Running LHC-MR on all pairs of traits</h3>
<p>First, please put the path to the folder containing your data in the
<code>sourceGWAS</code> object, and a <code>ListeofTraits</code> vector
with all your traits.<br />
The <code>AllPairs</code> object will contain all possible unique pairs
of traits.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="co"># sourceGWAS &lt;- &quot;GWAS_data/&quot;</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="co"># ListofTraits &lt;- c(&quot;LDL&quot;, &quot;HDL&quot;, &quot;CAD&quot;)</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>AllPairs <span class="ot">&lt;-</span> (<span class="fu">t</span>(<span class="fu">combn</span>(ListofTraits,<span class="dv">2</span>)))</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="fu">colnames</span>(AllPairs) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;Y&quot;</span>)</span></code></pre></div>
<p>Now, you can process your traits with LHC-MR. The function below is
simply a loop of the example from their github.<br />
The <code>NbCores</code> parameter is passed to <code>lhc_mr()</code>
and is very important to speed up calculation. Beware, a high number of
cores will greatly speed up the process, but will also consume a lot of
RAM.<br />
Results will be written in the <code>LHCMR_Results</code>folder.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>  </span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">library</span>(data.table)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  <span class="fu">library</span>(GenomicSEM)</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  <span class="fu">library</span>(TwoSampleMR)</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>  <span class="fu">library</span>(lhcMR)</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>  RefFolder <span class="ot">&lt;-</span> <span class="fu">getwd</span>()</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>  VAR <span class="ot">=</span> <span class="fu">fread</span>(<span class="fu">paste0</span>(RefFolder, <span class="st">&quot;/Necessary_data/variants.tsv.bgz&quot;</span>),</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>            <span class="at">select =</span> <span class="fu">c</span>(<span class="st">&quot;variant&quot;</span>, <span class="st">&quot;chr&quot;</span>, <span class="st">&quot;pos&quot;</span>, <span class="st">&quot;ref&quot;</span>, <span class="st">&quot;alt&quot;</span>, <span class="st">&quot;rsid&quot;</span>))</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>  VAR<span class="sc">$</span>chr<span class="ot">=</span><span class="fu">as.numeric</span>(VAR<span class="sc">$</span>chr)</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>lhcmrLoop <span class="ot">&lt;-</span> <span class="cf">function</span>(A, NbCores, <span class="at">Minimum_MAF =</span> <span class="fl">0.05</span>, <span class="at">run_ldsc =</span> T, <span class="at">run_MR =</span> T){</span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>  <span class="fu">gc</span>()</span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">&quot;Loop&quot;</span>,<span class="st">&quot; &quot;</span>,A)) </span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a>    debut <span class="ot">&lt;-</span> <span class="fu">Sys.time</span>()</span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a>  Trait1 <span class="ot">&lt;-</span> AllPairs[A,<span class="st">&quot;X&quot;</span>] <span class="co"># Get the trait from the specific pairing</span></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a>  Trait2 <span class="ot">&lt;-</span> AllPairs[A,<span class="st">&quot;Y&quot;</span>]</span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a></span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a>  </span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a>  <span class="co"># Import both sets of summary stats and remove missing observations (= variants with no calculable P-values)</span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a>  </span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a>  path1 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(sourceGWAS, <span class="fu">list.files</span>(sourceGWAS, <span class="at">pattern =</span> <span class="fu">paste0</span>(<span class="st">&quot;^&quot;</span>,Trait1,<span class="st">&quot;.csv&quot;</span>)))</span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a>  X <span class="ot">&lt;-</span> <span class="fu">fread</span>(path1)</span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a> </span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a>  path2 <span class="ot">&lt;-</span> <span class="fu">paste0</span>(sourceGWAS, <span class="fu">list.files</span>(sourceGWAS, <span class="at">pattern =</span> <span class="fu">paste0</span>(<span class="st">&quot;^&quot;</span>,Trait2,<span class="st">&quot;.csv&quot;</span>)))</span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a>  Y <span class="ot">&lt;-</span> <span class="fu">fread</span>(path2)</span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a> </span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a></span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a></span>
<span id="cb7-34"><a href="#cb7-34" tabindex="-1"></a>  </span>
<span id="cb7-35"><a href="#cb7-35" tabindex="-1"></a>  <span class="co"># Low confidence variants and variants with low minor allele frequencies (MAF) are filtered out.</span></span>
<span id="cb7-36"><a href="#cb7-36" tabindex="-1"></a>  X <span class="ot">&lt;-</span> X[ (<span class="sc">!</span> <span class="fu">is.na</span>(X<span class="sc">$</span>pval)) <span class="sc">&amp;</span> (<span class="sc">!</span> X<span class="sc">$</span>low_confidence_variant) <span class="sc">&amp;</span> ( X<span class="sc">$</span>minor_AF <span class="sc">&gt;</span> Minimum_MAF), ]</span>
<span id="cb7-37"><a href="#cb7-37" tabindex="-1"></a>  Y <span class="ot">&lt;-</span> Y[ (<span class="sc">!</span> <span class="fu">is.na</span>(Y<span class="sc">$</span>pval)) <span class="sc">&amp;</span> (<span class="sc">!</span> Y<span class="sc">$</span>low_confidence_variant) <span class="sc">&amp;</span> ( Y<span class="sc">$</span>minor_AF <span class="sc">&gt;</span> Minimum_MAF), ] </span>
<span id="cb7-38"><a href="#cb7-38" tabindex="-1"></a>  <span class="co">#</span></span>
<span id="cb7-39"><a href="#cb7-39" tabindex="-1"></a>  </span>
<span id="cb7-40"><a href="#cb7-40" tabindex="-1"></a>  </span>
<span id="cb7-41"><a href="#cb7-41" tabindex="-1"></a>  <span class="do">## To make sure we have all the columns needed (including allele data), we merge with Neale-provided variants file</span></span>
<span id="cb7-42"><a href="#cb7-42" tabindex="-1"></a></span>
<span id="cb7-43"><a href="#cb7-43" tabindex="-1"></a>  X <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(X,VAR[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)])</span>
<span id="cb7-44"><a href="#cb7-44" tabindex="-1"></a>  Y <span class="ot">=</span> dplyr<span class="sc">::</span><span class="fu">inner_join</span>(Y,VAR[,<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)])</span>
<span id="cb7-45"><a href="#cb7-45" tabindex="-1"></a>  </span>
<span id="cb7-46"><a href="#cb7-46" tabindex="-1"></a>  <span class="do">## File paths needed for the analysis</span></span>
<span id="cb7-47"><a href="#cb7-47" tabindex="-1"></a>  LD.filepath <span class="ot">=</span> <span class="fu">paste0</span>(RefFolder, <span class="st">&quot;/Necessary_data/LDscores_filtered.csv&quot;</span>) <span class="co"># LD scores</span></span>
<span id="cb7-48"><a href="#cb7-48" tabindex="-1"></a>  rho.filepath <span class="ot">=</span> <span class="fu">paste0</span>(RefFolder, <span class="st">&quot;/Necessary_data/LD_GM2_2prm.csv&quot;</span>) <span class="co"># local/SNP-specific LD scores</span></span>
<span id="cb7-49"><a href="#cb7-49" tabindex="-1"></a>  </span>
<span id="cb7-50"><a href="#cb7-50" tabindex="-1"></a>  </span>
<span id="cb7-51"><a href="#cb7-51" tabindex="-1"></a>  </span>
<span id="cb7-52"><a href="#cb7-52" tabindex="-1"></a>  </span>
<span id="cb7-53"><a href="#cb7-53" tabindex="-1"></a>  </span>
<span id="cb7-54"><a href="#cb7-54" tabindex="-1"></a>  ld <span class="ot">=</span> <span class="fu">paste0</span>(RefFolder, <span class="st">&quot;/Necessary_data/eur_w_ld_chr/&quot;</span>)  <span class="co">#LD information</span></span>
<span id="cb7-55"><a href="#cb7-55" tabindex="-1"></a>  hm3 <span class="ot">=</span> <span class="fu">paste0</span>(RefFolder, <span class="st">&quot;/Necessary_data/w_hm3.snplist&quot;</span>)</span>
<span id="cb7-56"><a href="#cb7-56" tabindex="-1"></a>  </span>
<span id="cb7-57"><a href="#cb7-57" tabindex="-1"></a> </span>
<span id="cb7-58"><a href="#cb7-58" tabindex="-1"></a>  <span class="do">## Step 1</span></span>
<span id="cb7-59"><a href="#cb7-59" tabindex="-1"></a>  trait.names<span class="ot">=</span><span class="fu">c</span>(Trait1,Trait2)</span>
<span id="cb7-60"><a href="#cb7-60" tabindex="-1"></a>  input.files <span class="ot">=</span> <span class="fu">list</span>(X,Y)</span>
<span id="cb7-61"><a href="#cb7-61" tabindex="-1"></a>  </span>
<span id="cb7-62"><a href="#cb7-62" tabindex="-1"></a>  <span class="fu">setwd</span>(<span class="st">&quot;LHCMR_Results&quot;</span>) <span class="co">#LHC-MR saves a lot of stuff in the working directory</span></span>
<span id="cb7-63"><a href="#cb7-63" tabindex="-1"></a>  </span>
<span id="cb7-64"><a href="#cb7-64" tabindex="-1"></a>  df <span class="ot">=</span> <span class="fu">merge_sumstats</span>(input.files,trait.names,LD.filepath,rho.filepath) <span class="co">#code from LHCMR</span></span>
<span id="cb7-65"><a href="#cb7-65" tabindex="-1"></a>  <span class="co">#save(df, file = &quot;gen_data&quot;)</span></span>
<span id="cb7-66"><a href="#cb7-66" tabindex="-1"></a>  <span class="fu">rm</span>(X)</span>
<span id="cb7-67"><a href="#cb7-67" tabindex="-1"></a>  <span class="fu">rm</span>(Y)</span>
<span id="cb7-68"><a href="#cb7-68" tabindex="-1"></a>  <span class="fu">rm</span>(input.files)</span>
<span id="cb7-69"><a href="#cb7-69" tabindex="-1"></a>  <span class="fu">gc</span>()</span>
<span id="cb7-70"><a href="#cb7-70" tabindex="-1"></a>  </span>
<span id="cb7-71"><a href="#cb7-71" tabindex="-1"></a>  <span class="do">## Step 2</span></span>
<span id="cb7-72"><a href="#cb7-72" tabindex="-1"></a>  SP_list <span class="ot">=</span> <span class="fu">calculate_SP</span>(df,trait.names,run_ldsc,run_MR,<span class="at">hm3=</span>hm3,<span class="at">ld=</span>ld,<span class="at">nStep =</span> <span class="dv">2</span>,</span>
<span id="cb7-73"><a href="#cb7-73" tabindex="-1"></a>                         <span class="at">SP_single=</span><span class="dv">3</span>,<span class="at">SP_pair=</span><span class="dv">100</span>,<span class="at">SNP_filter=</span><span class="dv">10</span>, <span class="at">nCores =</span> NbCores) <span class="co">#Calculating the starting points</span></span>
<span id="cb7-74"><a href="#cb7-74" tabindex="-1"></a>  </span>
<span id="cb7-75"><a href="#cb7-75" tabindex="-1"></a>  <span class="do">## Step 3</span></span>
<span id="cb7-76"><a href="#cb7-76" tabindex="-1"></a>  res <span class="ot">=</span> <span class="fu">lhc_mr</span>(SP_list, trait.names, <span class="at">paral_method=</span> <span class="st">&quot;lapply&quot;</span>, <span class="at">nBlock=</span><span class="dv">200</span>, </span>
<span id="cb7-77"><a href="#cb7-77" tabindex="-1"></a>               <span class="at">nCores =</span> NbCores, run_ldsc, run_MR) <span class="co">#The actual optimisation</span></span>
<span id="cb7-78"><a href="#cb7-78" tabindex="-1"></a>  </span>
<span id="cb7-79"><a href="#cb7-79" tabindex="-1"></a>  Nom <span class="ot">=</span> <span class="fu">paste0</span>(<span class="st">&quot;ResultsLHCMR_&quot;</span>, Trait1, <span class="st">&quot;_&quot;</span>, Trait2)</span>
<span id="cb7-80"><a href="#cb7-80" tabindex="-1"></a>  <span class="fu">save</span>(res, <span class="at">file =</span> Nom)</span>
<span id="cb7-81"><a href="#cb7-81" tabindex="-1"></a>  </span>
<span id="cb7-82"><a href="#cb7-82" tabindex="-1"></a>  </span>
<span id="cb7-83"><a href="#cb7-83" tabindex="-1"></a>  fin <span class="ot">&lt;-</span> <span class="fu">Sys.time</span>()</span>
<span id="cb7-84"><a href="#cb7-84" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">c</span>(Trait1,Trait2))</span>
<span id="cb7-85"><a href="#cb7-85" tabindex="-1"></a>  <span class="fu">print</span>(fin<span class="sc">-</span>debut)</span>
<span id="cb7-86"><a href="#cb7-86" tabindex="-1"></a>  </span>
<span id="cb7-87"><a href="#cb7-87" tabindex="-1"></a>  <span class="fu">setwd</span>(RefFolder)</span>
<span id="cb7-88"><a href="#cb7-88" tabindex="-1"></a></span>
<span id="cb7-89"><a href="#cb7-89" tabindex="-1"></a>  </span>
<span id="cb7-90"><a href="#cb7-90" tabindex="-1"></a>  <span class="fu">return</span>(res)</span>
<span id="cb7-91"><a href="#cb7-91" tabindex="-1"></a>}</span>
<span id="cb7-92"><a href="#cb7-92" tabindex="-1"></a></span>
<span id="cb7-93"><a href="#cb7-93" tabindex="-1"></a></span>
<span id="cb7-94"><a href="#cb7-94" tabindex="-1"></a>Range <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(AllPairs))  </span>
<span id="cb7-95"><a href="#cb7-95" tabindex="-1"></a></span>
<span id="cb7-96"><a href="#cb7-96" tabindex="-1"></a><span class="fu">lapply</span>(<span class="at">X =</span> Range, <span class="at">FUN =</span> lhcmrLoop, <span class="at">NbCores =</span> <span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="running-pleiovar" class="section level3">
<h3>Running PleioVar</h3>
<p>Let’s format the LHC-MR results in a PleioVar-friendly format, and
gather the last necessary inputs.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="fu">library</span>(stats)</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="fu">library</span>(parallel)</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="fu">library</span>(matrixStats)</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="fu">library</span>(PleioVar)</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a></span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>ParametersTable <span class="ot">&lt;-</span> <span class="fu">Format_parameters_table</span>(AllPairs, <span class="at">resdir =</span> <span class="st">&quot;LHCMR_Results&quot;</span>)</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>Index <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;Necessary_data/Index_FULL.csv&quot;</span>, <span class="at">header =</span> T, <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a><span class="fu">colnames</span>(Index) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;rsid&quot;</span>, <span class="st">&quot;variant&quot;</span>, <span class="st">&quot;LDscore&quot;</span>)</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>ListofTraits <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">c</span>(ParametersTable<span class="sc">$</span>X, ParametersTable<span class="sc">$</span>Y))</span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a>  </span></code></pre></div>
<p>The final step is to run the PleioVar pipeline.<br />
If you have a Zscore column in your GWAS data, PleioVar will process
them.<br />
If you do not, the Zscores will be computed with <span class="math inline">\(\frac{tstat}{\sqrt{(sample \: size)}}\)</span>. If
you formatted your data according to LHC-MR guidelines and the previous
steps worked, everything should run automatically.<br />
As usual, a higher number of cores in the <code>NbCores</code>
parameters will speed up calculation.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="fu">PleioVar_main</span>(ListofTraits, ParametersTable, Index , <span class="at">sourceGWAS =</span> sourceGWAS,</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>              <span class="at">NbCores =</span> <span class="dv">1</span>, <span class="at">gzip =</span> F, <span class="at">keepIntermediateFiles =</span> F)</span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
